python/tritonserver/_api/_model.py

```diff
@@ -180,40 +181,52 @@ def async_infer(
         if inference_request is None:
             inference_request = InferenceRequest(model=self, **kwargs)
 
-        if (inference_request.response_queue is not None) and (
-            not isinstance(inference_request.response_queue, asyncio.Queue)
-        ):
-            raise InvalidArgumentError(
-                "asyncio.Queue must be used for async response iterator"
-            )
-
+        # The 'request' is referencing pointers managed by the 'inference_request', so
+        # the 'inference_request' needs to be kept alive as long as the 'request' is
+        # alive.
         request = inference_request._create_tritonserver_inference_request()
 
-        response_iterator = AsyncResponseIterator(
-            self,
-            request,
-            inference_request.response_queue,
-            raise_on_error,
-        )
+        self._server.infer_async(request)
 
-        response_allocator = ResponseAllocator(
-            inference_request.output_memory_allocator,
-            inference_request.output_memory_type,
-        ).create_tritonserver_response_allocator()
+        class AsyncResponseIterator:
```

Comment:
I see this is a nested class - is that needed?

Reply:
it is not needed, the iterators are moved to the _response.py.